src/utils/isReactComponent.js

```diff
@@ -1,14 +1,16 @@
 export default function isReactComponent (component) {
-  if (typeof component === 'object') {
+  if (typeof component === 'object' && !isStyledComponent(component)) {
     return false
-  } else if (
+  }
+
+  return !(
     typeof component === 'function' &&
     component.prototype &&
     component.prototype.constructor.super &&
     component.prototype.constructor.super.name.startsWith('Vue')
-  ) {
-    return false
-  } else {
-    return true
-  }
+  )
+}
+
+function isStyledComponent (component) {
```

Comment:
We're not checking for a styled component, we're checking for React ref forwarding. So this function should be named something like `isReactForwardedRef`.